pkg/controller/pod/pod_controller.go

```diff
@@ -169,18 +174,31 @@ func (r *ReconcilePod) isForNsxNcpResource(request reconcile.Request) bool {
 	return false
 }
 
+func (r *ReconcilePod) isForNsxNodeAgentPod(request reconcile.Request) bool {
+	if (request.Namespace == operatortypes.NsxNamespace && strings.Contains(
+		request.Name, operatortypes.NsxNodeAgentDsName) &&
+		request.Name != operatortypes.NsxNodeAgentDsName) {
+		return true
+	}
+	return false
+}
+
 // Reconcile updates the ClusterOperator.Status to match the current state of the
 // watched Deployments/DaemonSets
 func (r *ReconcilePod) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 
 	if !r.isForNsxNcpResource(request) {
```

Comment:
renamed. But I think we shouldn't move the block for firstBoot. Because the firstBoot is for a corner case: nsx-node-agent becomes unhealthy -> NetworkUnavailable=True -> operator off -> nsx-node-agent becomes healthy and keeps running -> operator up -> operator cannot receive nsx-node-agent event to set NetworkUnavailable=False. So we need to full sync the nsx-node-agent pod statuses once  the operator up, no matter whether the event is for ncp resources or not. Will add the comment in code to explain it.